src/app/tx/[tx]/MessageDisplays.tsx

```diff
@@ -0,0 +1,158 @@
+'use client';
+import dynamic from 'next/dynamic';
+import { ExternalLink } from '../../../components/ExternalLink';
+import {
+  depositMessageStatusDisplay,
+  getRetryableIdOrDepositHash,
+  l1ToL2MessageToStatusDisplay,
+} from '../../../lib';
+import {
+  L1ToL2MessagesAndDepositMessages,
+  MessageStatusDisplay,
+} from '../../../types';
+
+const getDataFromL1ToL2Message = (
```

Comment:
Maybe move `getDataFromL1ToL2Message` to `lib` or `utils`?

Reply:
This one is only parsing data, and we only use it in this file, that's why I kept it in this file